test script panic

The test script threw a panic. The version of Go is 1.13.4 linux/amd64.
The code and the error info are as follows. It seems that there is something wrong with the internal cache.

```golang
package main

import (
	"bytes"
	"crypto/sha256"
	"encoding/binary"
	"fmt"
	"log"
	"time"

	"github.com/cockroachdb/pebble"
	"github.com/cockroachdb/pebble/bloom"
)

func main() {
	level := pebble.LevelOptions{
		FilterPolicy: bloom.FilterPolicy(10),
	}
	db, err := pebble.Open("/home/yann/test/pebble", &pebble.Options{
		BytesPerSync: 4 << 20,
		Levels:       []pebble.LevelOptions{level},
	})
	if err != nil {
		log.Fatal(err)
	}

	testValue := []byte("giu398egfisugdiwegfaiulfffffffffffglgieruiugfiuadgugfuihfd3ugfogeiof;hdfheodgfeugfdj;gf;3orfigfdagfuiehrfuiehiafbdjafhariugfhuaiegfriuhfeagauiffaeferhi;hgoaigf")
	start := time.Now()
	for i := 0; i < 1e8; i++ {
		fmt.Println(i)
		sum := sha256.Sum256(IntToBytes(i))
		key := sum[:]
		if err := db.Set(key, testValue, pebble.NoSync); err != nil {
			log.Fatal(err)
		}
	}
	fmt.Println("time cost:", time.Since(start))

	if err := db.Close(); err != nil {
		log.Fatal(err)
	}
}

func IntToBytes(n int) []byte {
	data := int64(n)
	bytebuf := bytes.NewBuffer([]byte{})
	binary.Write(bytebuf, binary.BigEndian, data)
	return bytebuf.Bytes()
}
```

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x885490]

goroutine 88 [running]:
github.com/cockroachdb/pebble/internal/cache.(*entry).unlink(...)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/entry.go:112
github.com/cockroachdb/pebble/internal/cache.(*shard).metaDel(0xc00032c958, 0xc00030ecc0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:341 +0x180
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:469 +0x94
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:428 +0x92
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00032c958)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/internal/cache/clockpro.go:456 +0xef
created by github.com/cockroachdb/pebble.(*DB).maybeScheduleCompactionPicker
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201023014740-c592aa69e4fd/compaction.go:1499 +0x3f7
```

I always run from an empty DB and the panic occours every time between 70,000 and 90,000 iterations. But the panic log is little different every time. It depends. I have pasted another log below. 

By the way, it works well at the first some days. And it can not work a week later. Nothing was changed during the period. This is really weird.

```shell
......
......
82302
82303
82304
82305
82306
82307
panic: pebble: inconsistent reference count: -1

goroutine 69 [running]:
github.com/cockroachdb/pebble/internal/cache.(*refcnt).release(0xc00105e018, 0xec7800)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/refcnt_normal.go:38 +0xe2
github.com/cockroachdb/pebble/internal/cache.(*Value).release(0xc00105e000)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/value.go:43 +0x31
github.com/cockroachdb/pebble/internal/cache.(*entry).setValue(0xc0002c7860, 0x0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/entry.go:141 +0x55
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:415 +0xfa
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:428 +0x90
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandTest(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:456 +0xef
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandHot(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:434 +0xcb
github.com/cockroachdb/pebble/internal/cache.(*shard).runHandCold(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:428 +0x90
github.com/cockroachdb/pebble/internal/cache.(*shard).evict(0xc00010cac8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:402 +0x2d
github.com/cockroachdb/pebble/internal/cache.(*shard).metaAdd(0xc00010cac8, 0x2, 0x8, 0x36bd6, 0xc001096a20, 0xc001096a20)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:280 +0x2f
github.com/cockroachdb/pebble/internal/cache.(*shard).Set(0xc00010cac8, 0x2, 0x8, 0x36bd6, 0xc001a02000, 0x0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:141 +0x331
github.com/cockroachdb/pebble/internal/cache.(*Cache).Set(0xc0000aa280, 0x2, 0x8, 0x36bd6, 0xc001a02000, 0x44f)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/internal/cache/clockpro.go:658 +0x83
github.com/cockroachdb/pebble/sstable.(*Reader).readBlock(0xc000f36f00, 0x36bd6, 0x44a, 0x0, 0xc0001c0270, 0x5, 0x20, 0x20)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/sstable/reader.go:1899 +0x2c9
github.com/cockroachdb/pebble/sstable.(*singleLevelIterator).loadBlock(0xc0001c0000, 0xc0001c00f8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/sstable/reader.go:292 +0x15b
github.com/cockroachdb/pebble/sstable.(*singleLevelIterator).skipForward(0xc0001c0000, 0x0, 0x0, 0x0, 0x0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/sstable/reader.go:692 +0x4f
github.com/cockroachdb/pebble/sstable.(*singleLevelIterator).Next(0xc0001c0000, 0xc000f40640, 0x9f, 0xc001034b20, 0x8f56be)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/sstable/reader.go:660 +0x16c
github.com/cockroachdb/pebble/sstable.(*twoLevelCompactionIterator).Next(0xc0001ec6c0, 0xc0019fbf4b, 0x9f, 0x1, 0xc001034b80)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/sstable/reader.go:1285 +0x62
github.com/cockroachdb/pebble.(*mergingIter).nextEntry(0xc000432300, 0xc000f40640)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/merging_iter.go:498 +0x7c
github.com/cockroachdb/pebble.(*mergingIter).Next(0xc000432300, 0x9f, 0x9f, 0x28, 0x50)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/merging_iter.go:990 +0x75
github.com/cockroachdb/pebble.(*compactionIter).iterNext(0xc000f40780, 0xc001034cd8)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction_iter.go:392 +0x38
github.com/cockroachdb/pebble.(*compactionIter).nextInStripe(0xc000f40780, 0x0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction_iter.go:413 +0x2f
github.com/cockroachdb/pebble.(*compactionIter).skipInStripe(0xc000f40780)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction_iter.go:377 +0x39
github.com/cockroachdb/pebble.(*compactionIter).Next(0xc000f40780, 0xc000fe4e80, 0x20, 0x20, 0x1)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction_iter.go:247 +0x7e
github.com/cockroachdb/pebble.(*DB).runCompaction(0xc0000b3000, 0xe, 0xc000f36c80, 0xb0dbe0, 0xef1240, 0xc0000aa460, 0xc0001201b0, 0x1, 0x1, 0x0, ...)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction.go:2253 +0x1489
github.com/cockroachdb/pebble.(*DB).compact1(0xc0000b3000, 0xc000f36c80, 0x0, 0x0, 0x0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction.go:1763 +0x167
github.com/cockroachdb/pebble.(*DB).compact.func1(0xb1ad60, 0xc000ffc8a0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction.go:1724 +0xbe
runtime/pprof.Do(0xb1ad60, 0xc000ffc8a0, 0xc00000cbe0, 0x1, 0x1, 0xc000f78798)
	/usr/local/go/src/runtime/pprof/runtime.go:37 +0xf8
github.com/cockroachdb/pebble.(*DB).compact(0xc0000b3000, 0xc000f36c80, 0x0)
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction.go:1721 +0xa2
created by github.com/cockroachdb/pebble.(*DB).maybeScheduleCompactionPicker
	/home/yann/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20201123184641-2da62788eede/compaction.go:1517 +0x3f7
exit status 2
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -30,6 +30,10 @@ matrix:
       go: 1.15.x
       os: linux
       script: make test TAGS=
+    - name: "go1.15.x-linux-no-cgo"
+      go: 1.15.x
+      os: linux
+      script: CGO_ENABLED=0 make test TAGS=
     - name: "go1.15.x-darwin"
       go: 1.15.x
       os: osx

--- a/internal/cache/cgo_disabled.go
+++ b/internal/cache/cgo_disabled.go
@@ -1,0 +1,9 @@
+// Copyright 2020 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+// +build !cgo
+
+package cache
+
+const cgoEnabled = false

--- a/internal/cache/cgo_enabled.go
+++ b/internal/cache/cgo_enabled.go
@@ -1,0 +1,9 @@
+// Copyright 2020 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+// +build cgo
+
+package cache
+
+const cgoEnabled = true

--- a/internal/cache/entry_normal.go
+++ b/internal/cache/entry_normal.go
@@ -21,7 +21,11 @@ const (
 	// Avoid using runtime.SetFinalizer in race builds as finalizers tickle a bug
 	// in the Go race detector in go1.15 and earlier versions. This requires that
 	// entries are Go allocated rather than manually allocated.
-	entriesGoAllocated = invariants.RaceEnabled
+	//
+	// If cgo is disabled we need to allocate the entries using the Go allocator
+	// and is violates the Go GC rules to put Go pointers (such as the entry
+	// pointer fields) into untyped memory (i.e. a []byte).
+	entriesGoAllocated = invariants.RaceEnabled || !cgoEnabled
 )
 
 var entryAllocPool = sync.Pool{

--- a/internal/cache/value_normal.go
+++ b/internal/cache/value_normal.go
@@ -18,6 +18,15 @@ func newValue(n int) *Value {
 	if n == 0 {
 		return nil
 	}
+
+	if !cgoEnabled {
+		// If Cgo is disabled then all memory is allocated from the Go heap and we
+		// can't play the trick below to combine the Value and buffer allocation.
+		v := &Value{buf: make([]byte, n)}
+		v.ref.init(1)
+		return v
+	}
+
 	// When we're not performing leak detection, the lifetime of the returned
 	// Value is exactly the lifetime of the backing buffer and we can manually
 	// allocate both.
@@ -34,6 +43,10 @@ func newValue(n int) *Value {
 }
 
 func (v *Value) free() {
+	if !cgoEnabled {
+		return
+	}
+
 	// When we're not performing leak detection, the Value and buffer were
 	// allocated contiguously.
 	n := valueSize + cap(v.buf)

